backend/src/flotilla/api/reports_api.py

```diff
@@ -0,0 +1,71 @@
+from http import HTTPStatus
+from logging import getLogger
+from typing import List
+
+from fastapi import APIRouter, Depends, Path, Response, Security
+from flotilla_openapi.models.problem_details import ProblemDetails
+from flotilla_openapi.models.report import Report
+from pytest import Session
+
+from flotilla.api.authentication import authentication_scheme
+from flotilla.database.crud import DBException, read_report_by_id, read_reports
+from flotilla.database.db import get_db
+from flotilla.database.models import ReportDBModel
+
+router = APIRouter()
+
+logger = getLogger("api")
+
+NOT_FOUND_DESCRIPTION = "Not Found - No report with given id"
+
+
+@router.get(
+    "/reports",
+    responses={
+        HTTPStatus.OK.value: {
+            "model": List[Report],
+            "description": "Request successful",
+        },
+    },
+    tags=["Reports"],
+    summary="List all available reports on the asset",
+    dependencies=[Security(authentication_scheme)],
+)
+async def get_reports(
+    db: Session = Depends(get_db),
+) -> List[Report]:
+    """List all available reports on the asset."""
+    db_reports: List[ReportDBModel] = read_reports(db)
+    logger.info(f"{db_reports}")
```

Comment:
Probably not necessary to log both the reports in the database and the returned reports. Suggest that both are removed until we see the need

Reply:
You are right, this is dev logging that I forgot to remove